services/sysinfo/client/client.go

```diff
@@ -187,3 +193,159 @@ func (p *dmesgCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interfa
+		  https://github.com/Snowflake-Labs/sansshell/blob/989cb789586532eaa22b75303ea94c97ca246306/services/util/util.go#L161.
+		  So It's best to reduce the the number of journal entries return based on these filtering flags.
+
+`
+}
+
+func (p *journalCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&p.since, "since", "", "Sets the date (YYYY-MM-DD HH:MM:SS) we want to filter from")
+	f.StringVar(&p.since, "S", "", "Sets the date (YYYY-MM-DD HH:MM:SS) we want to filter from (the date time is included)")
+	f.StringVar(&p.until, "until", "", "Sets the date (YYYY-MM-DD HH:MM:SS) we want to filter until (the date time is not included)")
+	f.StringVar(&p.until, "U", "", "Sets the date (YYYY-MM-DD HH:MM:SS) we want to filter until")
+	f.StringVar(&p.unit, "unit", "", "Sets systemd unit to filter messages")
+	f.StringVar(&p.output, "output", "", "Sets the format of the journal entries that will be shown. Right now only json and json-pretty are supported.")
+	f.Uint64Var(&p.tail, "tail", 100, "If positive, the latest n records to fetch. By default, fetch latest 100 records.")
+}
+
+func (p *journalCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	state := args[0].(*util.ExecuteState)
+	c := pb.NewSysInfoClientProxy(state.Conn)
+
+	// the output is case insensitive
+	p.output = strings.ToLower(p.output)
+	// currently output can only be json or json-pretty
+	if p.output != "" && p.output != "json" && p.output != "json-pretty" {
+		fmt.Fprintln(os.Stderr, "cannot set output to other formats unless json or json-pretty")
+		return subcommands.ExitUsageError
+	}
+
+	req := &pb.JournalRequest{
+		TailLine: uint32(p.tail),
+		Unit:     p.unit,
+		Output:   p.output,
+	}
+	// Note: if timestamp is passed, don't forget to conver to UTC before sending the rpc request
+	expectedTimeFormat := "2006-01-02 15:04:05"
+	loc, err := time.LoadLocation("Local")
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "cannot get local location")
+		return subcommands.ExitUsageError
+	}
+	if p.since != "" {
+		sinceTime, err := time.ParseInLocation(expectedTimeFormat, p.since, loc)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "please specify correct time pattern YYYY-MM-DD HH:MM:SS")
+			return subcommands.ExitUsageError
+		}
+		req.TimeSince = timestamppb.New(sinceTime.UTC())
+	}
+
+	if p.until != "" {
+		untilTime, err := time.ParseInLocation(expectedTimeFormat, p.until, loc)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, "please specify correct time pattern YYYY-MM-DD HH:MM:SS")
+			return subcommands.ExitUsageError
+		}
+		req.TimeUntil = timestamppb.New(untilTime.UTC())
+	}
+
+	stream, err := c.JournalOneMany(ctx, req)
+	if err != nil {
+		// Emit this to every error file as it's not specific to a given target.
+		for _, e := range state.Err {
+			fmt.Fprintf(e, "All targets - could not info servers: %v\n", err)
+		}
+		return subcommands.ExitFailure
+	}
+
+	targetsDone := make(map[int]bool)
```

Comment:
are you using this as a set? I did not see you set it to anything but true. If so, you can use struct{} vs bool. struct{} does not take any memory

Reply:
this a pattern I use for streaming receiving from https://github.com/Snowflake-Labs/sansshell/pull/128: So in order to keep consistency, I will keep it for now and probably we can refactor in the future for memory efficiency.